lte/gateway/python/integ_tests/s1aptests/s1ap_utils.py

```diff
@@ -307,6 +314,114 @@ def detach(self, ue_id, reason_type, wait_for_s1_ctxt_release=True):
         with self._lock:
             del self._ue_ip_map[ue_id]
 
+    def _verify_dl_flow(self, dl_flow_rules=None):
+        # try at least 5 times before failing as gateway
+        # might take some time to install the flows in ovs
+
+        # Verify the total number of DL flows for this UE ip address
+        num_dl_flows = 1
+        for key, value in dl_flow_rules.items():
+            ipv4_src_addr = None
+            tcp_src_port = 0
+            ip_proto = 0
+            ue_ip_str = str(key)
+            # Set to 1 for the default bearer
+            total_num_dl_flows_to_be_verified = 1
+            for item in value:
+                for flow in item:
+                    if flow["direction"] == "DL":
+                        total_num_dl_flows_to_be_verified += 1
+            total_dl_ovs_flows_created = get_flows(
+                self.datapath,
+                {
+                    "table_id": self.SPGW_TABLE,
+                    "match": {
+                        "nw_dst": ue_ip_str,
+                        "eth_type": 2048,
+                        "in_port": self.LOCAL_PORT,
+                    },
+                },
+            )
+            assert (
+                len(total_dl_ovs_flows_created)
+                == total_num_dl_flows_to_be_verified
+            )
```

Comment:
why do we need this? We are verifying each DL flow rule one by one in this function.

Reply:
The DL flows that we are verifying is only for the dedicated bearers as the flow rules passed from the test script are created only for the dedicated bearer. The only way to verify if a rule is created for default bearer is to check the total number of DL flows